### ISSUE TITLE
Add `cbrt` function to the standard

### ISSUE THREAD

**Issue by steff456** [data-apis/array-api#590]
This RFC requests to include a new API in the array API specification for the purpose of computing the cube root.

## Overview

Based on array comparison [data](https://github.com/data-apis/array-api-comparison/blob/7ecfa9867ec43a28e914b35868c041443d31d772/data/common_apis.csv#L32), the API is available in the majority of libraries in the PyData ecosystem.

The Array API Specification currently includes `pow` and `sqrt`, but does not include the IEEE 754 function `cbrt`.

- pow: <https://github.com/data-apis/array-api/blob/3d918783ac7f11cb5b31681b2badddeca81dc71e/src/array_api_stubs/_2022_12/elementwise_functions.py#L1786>

- sqrt: <https://github.com/data-apis/array-api/blob/3d918783ac7f11cb5b31681b2badddeca81dc71e/src/array_api_stubs/_2022_12/elementwise_functions.py#L2139>

While the cube root could be implemented in terms of `pow`, this is not desirable as the rational 1/3 is not typically equal to `1.0/3.0` due to limited floating-point precision. Cube root implementations are generally more accurate than the equivalent operation via `pow`.

## Prior art

- NumPy: <https://numpy.org/doc/stable/reference/generated/numpy.cbrt.html>
- PyTorch: (not currently implemented)
- MXNet: <https://mxnet.apache.org/versions/master/api/python/docs/api/np/generated/mxnet.np.cbrt.html>
- TensorFlow: <https://www.tensorflow.org/api_docs/python/tf/experimental/numpy/cbrt>
- C99: <https://en.cppreference.com/w/c/numeric/math/cbrt>

## Proposal:

```python
def cbrt(x: array, /) -> array
```

**Comment by asmeurer** [comment]
It looks like the NumPy cbrt doesn't support complex arguments. 

Presumably for a floating-point negative argument cbrt should return a real negative number, but for a complex real negative argument it should return the principle cube root (if we add complex support).

**Comment by kgryte** [comment]
Complex cube root is tricky. My initial sense is that we'd want to leave unspecified.

**Comment by asmeurer** [comment]
Even without implementing the complex version there's still the fact that cbrt(-1) returning -1 is a non-principle cube root (differing from `power`):

```py
>>> np.power(-1, 1/3)
<stdin>:1: RuntimeWarning: invalid value encountered in power
nan
>>> np.cbrt(-1)
-1.0
```

**Comment by kgryte** [comment]
Agreed. For negative reals, `cbrt` should return a negative real number. And thanks for highlighting a key limitation of `power`.

**Comment by asmeurer** [comment]
I don't really see it as a limitation. There are very good reasons for power to return principle roots, even if that means returning nan instead of a real value for noncomplex floating-point inputs. 

Of course, sometimes you do want a real cube root. And it seems that that's the C standard for what `cbrt` does, so we should definitely do that. 

I agree that omitting an extension to complex numbers is probably best because that would require choosing between either returning different values for negative reals, or using a non-principle branch cut, and both seem problematic. (also do Probably worth bringing up in a consortium meeting though.

**Comment by kgryte** [comment]
Linking to a (closed) feature request on PyTorch: https://github.com/pytorch/pytorch/issues/25766
